src/atc/etl/transformer.py

```diff
@@ -1,49 +1,30 @@
 from abc import abstractmethod
-from typing import List, Dict, Union
 
 from pyspark.sql import DataFrame
 
+from atc.etl.types import dataset_group, EtlBase
 
-class Transformer:
-    @abstractmethod
-    def process(self, df: DataFrame) -> DataFrame:
-        return df
-
-
-class MultiInputTransformer:
-    @abstractmethod
-    def process_many(self, dataset: Dict[str, DataFrame]) -> DataFrame:
-        pass
 
+class Transformer(EtlBase):
+    """If you only want to transform a single input dataframe,
+    implement `process`
+    if you want to transform a set of dataframes,
+    implement `process_many`
 
-class DelegatingTransformer(Transformer):
-    def __init__(self, inner_transformers: List[Transformer]):
-        super().__init__()
-        self.inner_transformers = inner_transformers
+    In regards to the etl step, the transformer CONSUMES the inputs
+    and ADDs the result of its transformation stage.
+    """
 
-    def get_transformers(self) -> List[Transformer]:
-        return self.inner_transformers
+    def etl(self, inputs: dataset_group) -> dataset_group:
+        if len(inputs) == 1:
+            for v in inputs.values():
```

Comment:
A bit wired to have a for loop for dict with len = 1?
The same could be written this way: next(iter(d.values()))